src/child.cpp

```diff
@@ -0,0 +1,470 @@
+                    std::cout << LOG_WARNING << "QSCAND: No device specified!\n";
+                else
+#endif
+                    std::cout << "QSCAND: No device specified!\n";
+                _exit(0);
+            }
+            if ( (mode == scan) && !strlen(test)) {
+#ifdef DAEMON_EN
+                if (daemonized)
+                // TODO: Redirect cout statements to file to emulate syslog behavior
+                    std::cout << LOG_WARNING << "QSCAND: No test specified!\n";
+                else
+#endif
+                    std::cout << "QSCAND: No test specified!\n";
+                _exit(0);
+            }
+            argv = add_arg(argv, &argc, "qscan");
+            switch (mode) {
+                case scanbus:
+                    argv = add_arg(argv, &argc, "-l");
+                    break;
+                case scan:
+                    argv = add_arg(argv, &argc, "-d");
+                    argv = add_arg(argv, &argc, device);
+
+                    argv = add_arg(argv, &argc, "-t");
+                    argv = add_arg(argv, &argc, test);
+
+                    std::cout << speed;
+                    argv = add_arg(argv, &argc, "-s");
+                    argv = add_arg(argv, &argc, speeds);
+
+                    if (!strcmp(test, "wt") && !WT_simul)
+                        argv = add_arg(argv, &argc, "-W");
+                    break;
+                case dinfo:
+                    argv = add_arg(argv, &argc, "-d");
+                    argv = add_arg(argv, &argc, device);
+                    argv = add_arg(argv, &argc, "-Ip");
+                    break;
+                case minfo:
+                    argv = add_arg(argv, &argc, "-d");
+                    argv = add_arg(argv, &argc, device);
+                    argv = add_arg(argv, &argc, "-m");
+                    break;
+                case help:
+                    argv = add_arg(argv, &argc, "-h");
+                    break;
+                default:
+                    if (debug && !daemonized)
+                        std::cout << "unknown mode: " << mode << "\n";
+                    return;
+            }
+
+            if ((cpid = createChildProcess(argv, &pipefd, NULL)) == -1) {
+#ifdef DAEMON_EN
+                if (daemonized)
+                // TODO: Redirect cout statements to file to emulate syslog behavior
+                    std::cout << LOG_ERR << "Can't start child!\n";
+                else
+#endif
+                    std::cout << "QSCAND: Can't start child!\n";
+                return;
+            } else {
+                // parent. copy messages from pipe to socket 
+                close(pipefd[1]); // unused write end
+
+                std::cout << "QSCAND: child created, reading from pipe...\n";
+#ifdef _WIN32
+                send(arg->connfd, lineo, strlen(lineo), 0);
+#else
+                write(arg->connfd, lineo, strlen(lineo));
+#endif
+                int wn, woffs;
```

Comment:
Codacy found an issue: [The scope of the variable 'woffs' can be reduced.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)